app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -156,7 +156,7 @@ def pulp_mirror?
 
       def pulp_master?
         # use map instead of pluck in case the features aren't saved yet during create
-        self.features.map(&:name).include?(PULP_FEATURE)
+        self.smart_proxy_features.any? { |smart_proxy_feature| smart_proxy_feature.feature.try(:name).try(:eql?, PULP_FEATURE) }
```

Comment:
can you rewrite it to 
```
self.smart_proxy_features.any? { |smart_proxy_feature| PULP_FEATURE == smart_proxy_feature.feature.try(:name)}
```

Reply:
This code makes me think that smart_proxy_features isn't a relation since we're using 'any?' vs 'where' is that right? Why isn't it a relation? I *thought* that the features were a model object. Just wondering for my own knowledge

Reply:
@jturel so smart_proxy used to be related to features via HABTM relationship. That recently got changed here to has_many through -> https://github.com/theforeman/foreman/commit/9080d4e5bc4fe7c7628cc1fce738302aceaed7ce

since the association to org and locations happens "before_create" all the entries are not fully baked in the db yet. That is causing this side effect where ```self.smart_proxy_features``` returns a list of features while ```self.features   ``` turns up empty. So we need to resort to looking at smart_proxy_features model directly

Reply:
Got it. Thanks @parthaa !